config/tls.go

```diff
@@ -66,7 +68,9 @@ func (cfg *tlsOptions) toProtoConfig() *proto.TLSEndpoint {
 
 	opts.MutualTLSAtEdge = mutualTLSEndpointOption(cfg.MutualTLSCA).toProtoConfig()
 
-	if cfg.KeyPEM != nil && cfg.CertPEM != nil {
+	// When terminate-at-edge is set the TLSTermination must be sent even if the key and cert are nil,
+	// this will default to the ngrok edge's automatically provisioned keypair.
+	if cfg.terminateAtEdge || (cfg.KeyPEM != nil && cfg.CertPEM != nil) {
```

Comment:
Let's reduce this to just `if cfg.terminateAtEdge { ... }`, since eventually we may support in-library termination where we definitely don't want to send the cert/key ngrok-side.

Reply:
Updated to just use the bool.